python/pyspark/sql/dataframe.py

```diff
@@ -1446,7 +1457,7 @@ def all_of_(xs):
         if isinstance(to_replace, (float, int, long, basestring)):
             to_replace = [to_replace]
 
-        if isinstance(value, (float, int, long, basestring)):
+        if isinstance(value, (float, int, long, basestring)) or value is None:
```

Comment:
This looks causing the warning always:

```python
>>> df = sc.parallelize([("Alice", 1, 3.0)]).toDF()
>>> df.replace({"Alice": "Bob"}).show()
```
```
.../spark/python/pyspark/sql/dataframe.py:1466: UserWarning: to_replace is a dict and value is not None. value will be ignored.
  warnings.warn("to_replace is a dict and value is not None. value will be ignored.")
...
```

Could we put this line into 1468L (under `else`)?